climetlab/core/ipython.py

```diff
@@ -42,15 +42,15 @@ def guess_which_ipython():
     if ipython_active.ipython_dir == "/deepnote-config/ipython":
         return ("deepnote", None)
 
-    if ipython_active.ipython_dir == "/home/jovyan/.ipython":
```

Comment:
let's keep this one also, I am not sure that ```__class__.__name__== 'ZMQInteractiveShell':``` below would take care of all jupyter-lab environments.

Reply:
but the ipython_dir should be different in other environment. For example , in my environment the ipython_acitve.ipython_dir is '/home/longtsing/.ipython' .So I think  the code 
`if ipython_active.ipython_dir == "/home/jovyan/.ipython":` should be change to 
`ipython_dir.endswith('.ipython')`

Reply:
IMO when``ipython_active.ipython_dir == "/home/jovyan/.ipython"```, we are quite sure that we have jupyter-lab, this happens in jupyter-loab docker container and other environments. So this can stay here safely.

Your addition ```ipython_active.__class__.__name__== 'ZMQInteractiveShell':`` is also good to take care of other cases (such as your local install).

```ipython_dir.endswith('.ipython')``` is more general, less reliable and more fragile, it could trigger in collab, deepnote, depending on what they choose to do there.
We could still add it at the very end, I suppose.